VespaGAS.xcodeproj/project.pbxproj

```diff
@@ -8,6 +8,7 @@
 
```

Comment:
" I can't declare amount and unit in the protocol extension"

True and unfortunate. There is a solution, but I don't like it, you're not going to like it, and I'm sure @dominicfreeston doesn't like it. As you probably know, all `NSObject` instances have a dictionary property that you can add stuff to so you could add properties to an object at runtime, even from within a protocol. However, that means you'd have to give up the value nature of your types and make them classes (not just any classes either - they'd have to subclass `NSObject`). If you're sticking to value types, then I'm not aware of any way to get around the fact that you can't add stored property declarations to protocols.